mmdet/models/detectors/mask_text_spotter.py

```diff
@@ -0,0 +1,105 @@
+
+    def export(self, img, img_metas, f='', **kwargs):
+
+        def export_to_onnx_text_recognition_decoder(net, input_size, path_to_onnx):
+            net.eval()
+
+            dim = net.hidden_size
+            prev_input = np.random.randn(1).astype(np.float32)
+            pev_hidden = np.random.randn(1, 1, dim).astype(np.float32)
+            prev_cell = np.random.randn(1, 1, dim).astype(np.float32)
+            encoder_outputs = np.random.randn(1, input_size[0] * input_size[1], dim).astype(np.float32)
+            prev_input = torch.tensor(prev_input)
+            pev_hidden = torch.tensor(pev_hidden)
+            prev_cell = torch.tensor(prev_cell)
+            encoder_outputs = torch.tensor(encoder_outputs)
+            if torch.cuda.is_available():
+                net = net.cuda()
+                prev_input = prev_input.cuda()
+                pev_hidden = pev_hidden.cuda()
+                prev_cell = prev_cell.cuda()
+                encoder_outputs = encoder_outputs.cuda()
+            if isinstance(net.decoder, nn.GRU):
+                torch.onnx.export(net, (prev_input, pev_hidden, encoder_outputs),
+                                  path_to_onnx, verbose=True,
+                                  input_names=['prev_symbol', 'prev_hidden', 'encoder_outputs'],
+                                  output_names=['output', 'hidden', 'attention']
+                                  )
+            elif isinstance(net.decoder, nn.LSTM):
+                torch.onnx.export(net, (prev_input, pev_hidden, encoder_outputs, prev_cell),
+                                  path_to_onnx, verbose=True,
+                                  input_names=['prev_symbol', 'prev_hidden', 'encoder_outputs', 'prev_cell'],
+                                  output_names=['output', 'hidden', 'cell', 'attention']
+                                  )
+
+            printable_graph = onnx.helper.printable_graph(onnx.load(path_to_onnx).graph)
+            print(printable_graph)
+
+            return printable_graph
+
+        def export_to_onnx_text_recognition_encoder(net, input_size, path_to_onnx):
+            net.eval()
+            dim = net.dim_input
+            input = np.random.randn(1, dim, *input_size).astype(np.float32)
+            input = torch.tensor(input)
+            if torch.cuda.is_available():
+                net = net.cuda()
+                input = input.cuda()
+            input.requires_grad = True
+            torch.onnx.export(net, input, path_to_onnx, verbose=True,
+                              input_names=['input'], output_names=['output'])
+
+            printable_graph = onnx.helper.printable_graph(onnx.load(path_to_onnx).graph)
+            print(printable_graph)
+
+            return printable_graph
+
+        self.img_metas = img_metas
+        self.forward_backup = self.forward
+        self.forward = self.forward_export
+        torch.onnx.export(self, img, f, **kwargs)
+        self.forward = self.forward_backup
```

Comment:
Don't you have to call `torch.onnx.export` inside the `forward_export_context` ?

@LeonidBeynenson could you please clarify that.

Reply:
Yes, it is better to use my forward export context here.

Although I have to say that this function contains two subfunctions that make `torch.onnx.export` for subgraphs. 
We do not have special export contexts for the subgraphs at the moment (in the future we could do it to make integration with NNCF, but it should not be done now). 
So, anyway one forward export context won't help 🙂  
But one is better than nothing. And it will give us a hope that NNCF will work with this network too!

So, Ilya, please, use forward export context here.

Reply:
fixed